compilers/concrete-compiler/compiler/Makefile

```diff
@@ -1,7 +1,9 @@
 LLVM_PROJECT_DIR=../llvm-project
 
 BUILD_TYPE?=Release
-BUILD_DIR?=./build
+# Build dir is `build` for Release builds and `build_BUILD_TYPE` for 
+# other build types (e.g. Debug, ReleaseWithDebugInfo, etc.)
+BUILD_DIR?=./$(shell ./get_build_dir.sh "$(BUILD_TYPE)")
```

Comment:
This new script produces either:

`build` -> for Release builds
`build_$(BUILD_TYPE)` (e.g. build_Debug) for other build types